### ISSUE TITLE
Autoscaling signal has empty condition

### ISSUE THREAD

**Issue by sigismund** [hashicorp/terraform-provider-google#15033]
## TL;DR

When using the following autoscaling signal, an empty condition is created, which causes a UI error `This item contains one or more errors`

```hcl
 metric {
  name   = "compute.googleapis.com/instance/cpu/utilization"
  target = 0.75
  type   = "GAUGE"
}
```

### Expected behavior

Expected Metrics signal to look like this:

![image](https://github.com/hashicorp/terraform-provider-google/assets/3924612/53731eff-80cf-4866-bdfa-b082567b141a)

### Observed behavior

Instead, there is an empty condition that is causing a UI issue.

![image](https://github.com/hashicorp/terraform-provider-google/assets/3924612/fe302ec9-5b1e-48f9-9ff4-957266d62383)

### Terraform Configuration

```hcl
resource "google_compute_autoscaler" "default" {
  provider = google-beta

  name   = "my-autoscaler"
  zone   = "us-central1-f"
  target = google_compute_instance_group_manager.default.id

  autoscaling_policy {
    max_replicas    = 5
    min_replicas    = 1
    cooldown_period = 60

    metric {
      name   = "compute.googleapis.com/instance/cpu/utilization"
      target = 0.75
      type   = "GAUGE"
    }
  }
}

resource "google_compute_instance_template" "default" {
  provider = google-beta

  name           = "my-instance-template"
  machine_type   = "e2-medium"
  can_ip_forward = false

  tags = ["foo", "bar"]

  disk {
    source_image = data.google_compute_image.debian_9.id
  }

  network_interface {
    network = "default"
  }

  metadata = {
    foo = "bar"
  }

  service_account {
    scopes = ["userinfo-email", "compute-ro", "storage-ro"]
  }
}

resource "google_compute_target_pool" "default" {
  provider = google-beta

  name = "my-target-pool"
}

resource "google_compute_instance_group_manager" "default" {
  provider = google-beta

  name = "my-igm"
  zone = "us-central1-f"

  version {
    instance_template = google_compute_instance_template.default.id
    name              = "primary"
  }

  target_pools       = [google_compute_target_pool.default.id]
  base_instance_name = "autoscaler-sample"
}

data "google_compute_image" "debian_9" {
  provider = google-beta

  family  = "debian-11"
  project = "debian-cloud"
}

provider "google-beta" {
  region = "us-central1"
  zone   = "us-central1-a"

}
```

### Terraform Version

Terraform: v1.5.1

Google Provider: 4.50.0 & 4.64.0 (tested with 2 different versions)

### Additional information

Terraform might be accounting for the aggregator field and adding the extra = that is causing the issue. If i click edit from the UI and turn on and off the aggregator the extra condition is removed and the issue is fixed.

![image](https://github.com/hashicorp/terraform-provider-google/assets/3924612/50eee3cb-354b-4eed-89ed-aacb21c38e8b)

**Comment by edwardmedia** [comment]
@sigismund have you tried adding [filter](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_autoscaler#filter) in your config?

**Comment by sigismund** [comment]
@edwardmedia By adding basic filter (`filter = "metric.type=\"compute.googleapis.com/instance/cpu/utilization\" resource.type=\"gce_instance\""` ) I was able to bypass the error. So I guess the issues is related to the `filter`.

**Comment by edwardmedia** [comment]
@sigismund I think you need to provide settings for `filter` if you are looking for its specific values. Does this make sense?

**Comment by sigismund** [comment]
@edwardmedia Sorry, I think there was a misunderstanding. In our configuration, we are not using a filter. I was simply testing if adding a filter could bypass the error.

If you check the Terraform sample that I provided in the original description, you can see the configuration without a filter. Also, according to the [documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_autoscaler#filter) , the filter is an optional parameter for the google_compute_autoscaler resource. Therefore, the original configuration that I posted should not cause any issues.

**Comment by edwardmedia** [comment]
I see what you mean. @sigismund. I think we need to provide a [default](https://github.com/hashicorp/terraform-provider-google-beta/blob/265b41af3f9792066bfcbd746cf4751327143f82/google-beta/services/compute/resource_compute_autoscaler.go#L193C53-L193C53) value for filter

```
If not specified, the type defaults to gce_instance.
```

**Comment by c2thorn** [comment]
@zli82016 is there an upgrade guide for this?

**Comment by zli82016** [comment]
Yeah, the guide is merged to the main branch. https://github.com/GoogleCloudPlatform/magic-modules/pull/8648
